integration_tests/models/sql/schema.yml

```diff
@@ -182,3 +182,29 @@ models:
     tests:
       - dbt_utils.equality:
           compare_model: ref('data_deduplicate_expected')
+
+  - name: test_not_empty_string_failing
+    columns:
+      - name: string
+        tests:
+          - dbt_utils.not_empty_string:
+              trim_whitespace: True
+              config:
+                severity: error
+                error_if: "<1"
+                warn_if: "<0"
+          - dbt_utils.not_empty_string:
+              trim_whitespace: False
+              config:
+                severity: error
+                error_if: "<1"
+                warn_if: "<0"
+
+  - name: test_not_empty_string_passing
+    columns:
+      - name: string
+        tests:
+          - dbt_utils.not_empty_string:
+              trim_whitespace: True
+          - dbt_utils.not_empty_string:
+              trim_whitespace: False
```

Comment:
If you can run the same test on the same column with two different configs, is one of the configs redundant? This feels odd to me

Reply:
I totally see your point here. I was trying to also test the `trim_whitespace` option. I've removed the redundant ones for now, but let me know if you have another suggestion for testing the option.

Reply:
I _think_ you just need to test two different columns? One with whitespace that needs trimmed to pass, and one that doesn't.

Reply:
@joellabes brilliant as always~✨ Added a `string_trim_whitespace_false` column to `test_not_empty_string_passing` that would cause the test to fail if we weren't properly trimming whitespace :)